bundles/plugins/org.bonitasoft.studio.businessobject/src/org/bonitasoft/studio/businessobject/core/operation/DeployBDMOperation.java

```diff
@@ -70,6 +71,7 @@ public void setSession(final APISession session) {
      */
     @Override
     public void run(final IProgressMonitor monitor) throws InvocationTargetException, InterruptedException {
+        assert (!PlatformUtil.isHeadless());
```

Comment:
Becarefull with those assert. They will not fail if jvm is not launched with specific argument (-ea).
You may use a Preconditions.checkState() instead.